utils/jump_host.py

```diff
@@ -24,23 +71,66 @@ def get_identity_from_vault(self, jh):
         return identity
 
     def init_ssh_base_cmd(self):
+        user_host = '{}@{}'.format(self.user, self.hostname)
+
+        self.set_ssh_base_cmd([
+            'ssh',
+            '-o', 'StrictHostKeyChecking=yes',
+            '-o', 'UserKnownHostsFile={}'.format(self.known_hosts_file),
+            '-i', self.identity_file, '-p', str(self.port), user_host])
+
+    def init_ssh_session(self):
+        session = jumpssh.SSHSession(
+            self.hostname,
+            self.user,
+            private_key_file=self.identity_file,
+            port=self.port,
+        )
+        self.set_ssh_session(session)
+
+    def init_identity_files(self):
         self._identity_dir = tempfile.mkdtemp()
+
         identity_file = self._identity_dir + '/id'
         with open(identity_file, 'w') as f:
             f.write(self.identity)
         os.chmod(identity_file, 0o600)
-        user_host = '{}@{}'.format(self.user, self.hostname)
+        self.identity_file = identity_file
 
-        self.set_ssh_base_cmd([
-            'ssh', '-o', 'StrictHostKeyChecking=no',
-            '-o', 'UserKnownHostsFile=/dev/null', '-o', 'LogLevel=ERROR',
-            '-i', identity_file, '-p', self.port, user_host])
+        known_hosts_file = self._identity_dir + '/known_hosts'
+        with open(known_hosts_file, 'w') as f:
+            f.write(self.known_hosts)
+        os.chmod(known_hosts_file, 0o600)
+        self.known_hosts_file = known_hosts_file
 
     def get_ssh_base_cmd(self):
```

Comment:
maybe we can put here:

```
        user_host = '{}@{}'.format(self.user, self.hostname)

         return [
            'ssh',
            '-o', 'StrictHostKeyChecking=yes',
            '-o', 'UserKnownHostsFile={}'.format(self.known_hosts_file),
            '-i', self.identity_file, '-p', str(self.port), user_host]
```
and remove the init and the setter

Reply:
fixed